src/Mapping/Annotation/Blameable.php

```diff
@@ -2,23 +2,42 @@
 
 namespace Gedmo\Mapping\Annotation;
 
+use Attribute;
 use Doctrine\Common\Annotations\Annotation;
+use Doctrine\Common\Annotations\Annotation\NamedArgumentConstructor;
 
 /**
  * Blameable annotation for Blameable behavioral extension
  *
  * @Annotation
+ * @NamedArgumentConstructor()
  * @Target("PROPERTY")
  *
  * @author David Buchmann <mail@davidbu.ch>
  * @license MIT License (http://www.opensource.org/licenses/mit-license.php)
  */
-final class Blameable extends Annotation
+#[Attribute(Attribute::TARGET_PROPERTY)]
+final class Blameable
 {
     /** @var string */
     public $on = 'update';
     /** @var string|array */
     public $field;
     /** @var mixed */
     public $value;
+
+    /**
+     *
+     * @param string $on
+     * @param null|string|array $field
+     * @param mixed|null $value
+     *
+     * @return void
+     */
+    public function __construct($on = 'update', $field = null, $value = null)
```

Comment:
This is a breaking change (removing the `Annotation` parent and changing the constructor signature). It would require a major release. Is it necessary, or can we find a backwards-compatible option?

Reply:
It's technically a break, but in terms of usage it should be fine because of the `@NamedArgumentConstructor` annotation: https://www.doctrine-project.org/projects/doctrine-annotations/en/1.13/custom.html#optional-constructors-with-named-parameters

So there is no need of extending `Annotation` class anymore.

Reply:
It's a bit of a weird case.  From a purely PHP API perspective, it is 100% a B/C break:

- Class inheritance changes (`$foo instanceof Doctrine\Common\Annotations\Annotation` goes from passing to failing with this PR applied)
- Constructor signature changes

From a practical perspective, it really starts to get murky.  `Gedmo\Mapping\Driver\AnnotationDriverInterface` doesn't have a hard dependency on the `Doctrine\Common\Annotations\Reader` interface, so there isn't an assurance that the `doctrine/annotations` package is even being used to read the annotations.  Even if there were, if someone is implementing the interface themselves and not using the readers or parser from that package (`Doctrine\Common\Annotations\DocParser` is where the logic to deal with the `@NamedArgumentConstructor` annotation lies), that reader might not be able to construct the annotation classes with the changed constructor signatures.  Granted, the odds of someone using a custom annotation reader and NOT using their parser are pretty low, but even if Doctrine's interfaces were typehinted, you can't guarantee a reader is using that compatibility layer.

Reply:
Thanks for the explanation, so what we can do is to keep the `array $data = []` as first parameter of the constructor and add the rest after it.

I've re-added it in https://github.com/doctrine-extensions/DoctrineExtensions/pull/2253